Managers/PerformanceMan.cpp

```diff
@@ -0,0 +1,264 @@
+#include "PerformanceMan.h"
+#include "MovableMan.h"
+#include "FrameMan.h"
+#include "AudioMan.h"
+#include "Timer.h"
+
+#include "GUI.h"
+#include "AllegroBitmap.h"
+
+namespace RTE {
+
+	const std::string PerformanceMan::c_ClassName = "PerformanceMan";
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PerformanceMan::Clear() {
+		m_ShowPerfStats = false;
+		m_AdvancedPerfStats = true;
+		m_CurrentPing = 0;
+		m_FrameTimer = 0;
+		m_MSPFs.clear();
+		m_MSPFAverage = 0;
+		m_SimSpeed = 1.0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int PerformanceMan::Create() {
+		m_FrameTimer = new Timer();
+		m_Sample = 0;
+
+		for (unsigned short c = 0; c < PERF_COUNT; ++c) {
+			for (unsigned short i = 0; i < c_MaxSamples; ++i) {
+				m_PerfData[c][i] = 0;
+				m_PerfPercentages[c][i] = 0;
+			}
+			m_PerfMeasureStart[c] = 0;
+			m_PerfMeasureStop[c] = 0;
+		}
+		// Set up performance counter's names
+		m_PerfCounterNames[PERF_SIM_TOTAL] = "Total";
+		m_PerfCounterNames[PERF_ACTORS_PASS1] = "Act Travel";
+		m_PerfCounterNames[PERF_PARTICLES_PASS1] = "Prt Travel";
+		m_PerfCounterNames[PERF_ACTORS_PASS2] = "Act Update";
+		m_PerfCounterNames[PERF_PARTICLES_PASS2] = "Prt Update";
+		m_PerfCounterNames[PERF_ACTORS_AI] = "Act AI";
+		m_PerfCounterNames[PERF_ACTIVITY] = "Activity";
+
+		return 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PerformanceMan::Destroy() {
+		delete m_FrameTimer;
```

Comment:
Have to include `Timer.h` in the header.

Reply:
You have to include the header to do delete?? Didn't realize that!

Reply:
Well i'm deleting a `Timer` object how is it supposed to know what it is without me including the header?

Reply:
I mean it works when you're returning a timer object like
```
Timer someFunction();
```
But I guess it doesn't work if you're actually interacting with it, which deleting counts as. Anyway, yeah, if it can't be done without including Timer then that's fine, leave it here.

Reply:
Yeah it doesn't.
I'm already forward declaring `Timer` in the header, but I still need the actual include for it to work.